modules/tensor_mechanics/include/materials/ComputeRSphericalIncrementalStrain.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTERSPHERICALINCREMENTALSTRAIN_H
+#define COMPUTERSPHERICALINCREMENTALSTRAIN_H
+
+#include "ComputeIncrementalSmallStrain.h"
+
+/**
+ * ComputeRSphericalIncrementalStrain defines a strain increment only
+ * for small strains in 1D spherical symmetry geometries.  The strains in the
+ * polar and azimuthal directions are functions of the radial displacement.
+
+ */
+class ComputeRSphericalIncrementalStrain : public ComputeIncrementalSmallStrain
+{
+public:
+  ComputeRSphericalIncrementalStrain(const InputParameters & parameters);
+
+protected:
+  /// the old value of the first component of the displacements vector
+  const VariableValue & _disp_old_0;
+
+  virtual void computeTotalStrainIncrement(RankTwoTensor & total_strain_increment, unsigned int qp);
```

Comment:
Why do you need to pass in the qp here? Can't you just use `_qp` in this method?